File: GameClient/Assets/Scripts/Framework/UI/UIManager.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using UnityEngine.UI;

public class UIManager : Singleton<UIManager>
{
    private GameObject uiCanvasPrefab;
    public GameObject UICanvas { get; set; }
    List<ViewBase> _views = new List<ViewBase>();

    public void AddView(ViewBase view)
    {       
        _views.Add(view);
        view.Start();
    }

    public void RemoveView(ViewBase view)
    {
        _views.Remove(view);
        view.OnDestroy();
    }

    public void Start()
    {
        uiCanvasPrefab = Resources.Load("Prefabs/UI/UICanvas") as GameObject;
        UICanvas = Object.Instantiate(uiCanvasPrefab);
    }

    public void Update()
    {
        foreach (var view in _views)
            view.Update();
    }
}

File: GameClient/Assets/HotFix/Lua/MVC/UIManager.lua
local UIManager = class()

function UIManager:start(  )
	self.UICanvasPrefab = Resources.Load("Prefabs/UI/UICanvas")
	self.UICanvas = GameObject.Instantiate(self.UICanvasPrefab)
end

function UIManager:addView( view )
	view:start()
	self.currentView = view 
end

function UIManager:removeView( view )
	view:destroy()
	self.currentView = nil 
end

function UIManager:update(  )
	self.currentView:update()
end

return UIManager